ocaml/xapi/xapi_network.ml

```diff
@@ -329,3 +329,62 @@ let with_networks_attached_for_vm ~__context ?host ~vm f =
         error "Caught %s while detaching networks" (string_of_exn e)
     end;
     raise e
+
+let assert_can_add_purpose ~__context ~network ~current porpoise =
+  let sop (*string-of-porpoise*) = Record_util.network_purpose_to_string in
+  let reject str =
+    raise (Api_errors.Server_error (Api_errors.network_incompatible_purposes, [ (sop porpoise); str ]))
+  in
+  if (List.mem `himn current) then (
+    info "Existing network purpose %s is not compatible with any other purpose." (sop `himn);
+    reject "any"
```

Comment:
I think this should be `himn` because from the datamodel it seems that the second parameter of the error is the conflicting purpose, but maybe I misunderstood it.

Reply:
Well spotted.